linera-core/src/client/mod.rs

```diff
@@ -938,12 +925,30 @@ where
         #[cfg(with_metrics)]
         let _latency = metrics::PREPARE_CHAIN_LATENCY.measure_latency();
 
-        // Verify that our local storage contains enough history compared to the
-        // expected block height. Otherwise, download the missing history from the
-        // network.
-        let next_block_height = self.next_block_height();
+        let mut info = self.sync_until(self.next_block_height()).await?;
+
+        // If, during sync above, we've learned about new owners we need to check if the chain
+        // has been extended beyoned `self.next_block_height()`.
+        if self.state().has_other_owners(&info.manager.ownership) {
+            // For chains with any owner other than ourselves, we could be missing recent
+            // certificates created by other owners. Further synchronize blocks from the network.
+            // This is a best-effort that depends on network conditions.
+            let nodes = self.validator_nodes().await?;
+            info = self.synchronize_chain_state(&nodes, self.chain_id).await?;
+        }
+        self.update_from_info(&info);
+        Ok(info)
+    }
+
+    // Verify that our local storage contains enough history compared to the
```

Comment:
(Nit: We use the third person singular for function doc comments, like the standard library, i.e. "Verifies" instead of "Verify".)

Reply:
Done.